upb/def.c

```diff
@@ -458,6 +468,10 @@ int32_t upb_enumvaldef_number(const upb_enumvaldef *ev) {
   return ev->number;
 }
 
+uint32_t upb_enumvaldef_index(const upb_enumvaldef *ev) {
+  return ev - ev->enum_->values;
```

Comment:
subtracting a pointer from something two levels deep seems suspicious...  are you sure?  If so add a comment

Reply:
Added comment (and added one for `upb_oneofdef_index()` too, which does the same thing). I also changed `enum_` to `parent`: hopefully this will help clarify that we are reaching one level up, not two levels deep.